src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java

```diff
@@ -129,28 +132,40 @@ synchronized int insertRow(String tableName, List<Object> row) throws SQLExcepti
   @Override
   synchronized String readTables() {
     StringBuilder jsonResult = new StringBuilder();
-    final JSONFormat jsonFormat = new JSONFormat().header(false);
     jsonResult.append("{");
-    // Currently returning a json string response generated by jooq library formatJSON method
-    // Need to change this because currently maps are being stored as strings in JSON instead of
-    // objects
     String tableJson =
         super.tableNames.stream()
-            .map(
-                table -> {
-                  StringBuilder jsonForEachTable = new StringBuilder();
-                  jsonForEachTable.append("\"" + table + "\":");
-                  Result<Record> result = create.select().from(table).fetch();
-                  jsonForEachTable.append(result.formatJSON(jsonFormat));
-                  return jsonForEachTable.toString();
-                })
+            .map(f -> readTable(f))
             .collect(Collectors.joining(","));
     jsonResult.append(tableJson);
     jsonResult.append("}");
     LOG.debug("JSON Result - " + jsonResult);
     return jsonResult.toString();
   }
 
+  private synchronized String readTable(String tableName) {
+    final JSONFormat jsonFormat = new JSONFormat().header(false);
+    StringBuilder jsonForEachTable = new StringBuilder();
+    jsonForEachTable.append("\"" + tableName + "\":");
```

Comment:
Let's not try to hand roll json here and use a JSONObject or some library to do it.

Reply:
done